examples/ssh/pfsense_pftop.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * This driver extracts information for pftop command in pfsense system
+ * The communication protocol is SSH
+ * This driver create a dynamic table containing the result of the pftop command
+ * table columns are:
+ * %Id
+ * %PR
+ * %DIR
+ * %SRC
+ * %DEST
+ * %GW
+ * %STATE
+ * %AGE
+ * %EXP
+ * %PKTS
+ * %BYTES
+ * %AVG
+ * %RULE
+ * Tested under pfsense 2.6.0-RELEASE
+ */
+
+var _var = D.device.createVariable;
+
+var table = D.createTable(
+    "pfTop",
+    [
+        { label: "PR" },
+        { label: "DIR" },
+        { label: "SRC" },
+        { label: "DEST" },
+        { label: "GW" },
+        { label: "STATE" },
+        { label: "AGE" },
+        { label: "EXP" },
+        { label: "PKTS" },
+        { label: "BYTES" },
+        { label: "AVG"},
+        { label: "RULE" }
+    ]
+);
+
+var ssh_config = {
+    username: D.device.username(),
+    password: D.device.password(),
+    port: 22,
+    timeout: 30000
+};
+
+function checkSshError(err) {
+    if(err.message) console.error(err.message);
+    if(err.code == 5) D.failure(D.errorType.AUTHENTICATION_ERROR);
+    if(err.code == 255) D.failure(D.errorType.RESOURCE_UNAVAILABLE);
+    console.error(err);
+    D.failure(D.errorType.GENERIC_ERROR);
+}
+function exec_command(command, callback) {
+    var config = JSON.parse(JSON.stringify(ssh_config));
+    config.command = command;
+    D.device.sendSSHCommand(config, function (out, err) {
+        if(err) checkSshError(err);
+        callback(out.split("\n"));
+    });
+}
+
+function convertToK(count, unit) {
+    switch (unit) {
+    case "K": return count;
+    case "M": return count * 1000;
+    case "G": return count * 1000000;
+    }
+}
+
+/**
+* @remote_procedure
+* @label Validate Association
+* @documentation This procedure is used to validate if pftop command is executed successfully
+*/
+function validate() {
+    exec_command("pftop -v long -w 200 -o bytes 50", function () {
```

Comment:
Put the '50' in a constant to state what its purpose is (the top 50 records of the top command)